File: xamarin/Application.XForms.Service/Models/Database/Topic.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WindnTreesSEO.Models.Database
{
    public partial class Topic
    {
        [Key]
        [Column("UID")]
        public Guid Uid { get; set; }
        [Column("CategoryID")]
        public Guid CategoryId { get; set; }
        [Column("SkillLevelID")]
        public Guid? SkillLevelId { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime RecordTime { get; set; }
        [StringLength(100)]
        public string Menu { get; set; }
        [Required]
        [StringLength(200)]
        public string Subject { get; set; }
        [Column(TypeName = "ntext")]
        public string Description1 { get; set; }
        [Column(TypeName = "ntext")]
        public string Description2 { get; set; }
        public bool? Active { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? UpdateTime { get; set; }
        public byte[] RowVersion { get; set; }

        [ForeignKey(nameof(CategoryId))]
        [InverseProperty("Topic")]
        public virtual Category Category { get; set; }
        [ForeignKey(nameof(SkillLevelId))]
        [InverseProperty("Topic")]
        public virtual SkillLevel SkillLevel { get; set; }
    }
}

File: xamarin/Application.XForms.Service/wwwroot/js/Models/Topic.js
function Topic(data) {
    var instance = this;

    //required for row uniqueness 
    instance._datakey = data.Uid;

    instance.Uid = data.Uid;

    instance.CategoryId = ko.observable(data.CategoryId);
    instance.CategoryId.extend({
        required: true
    });

    instance.SkillLevelId = ko.observable(data.SkillLevelId);
    instance.SkillLevelId.extend({
        required: true
    });

    instance.RecordTime = ko.observable(data.RecordTime);
    instance.UpdateTime = ko.observable(data.UpdateTime);

    instance.Menu = ko.observable(data.Menu);
    instance.Menu.extend({
        required: true,
        maxLength: 100
    });

    instance.Subject = ko.observable(data.Subject);
    instance.Subject.extend({
        required: true,
        maxLength: 100
    });

    instance.Description1 = ko.observable(data.Description1);
    instance.Description1.extend({
        maxLength: 200
    });

    instance.Description2 = ko.observable(data.Description2);
    instance.Description2.extend({
        maxLength: 200
    });

    instance.Active = ko.observable(data.Active);

    instance.RowVersion = data.RowVersion;

    //category reference object
    instance.Category = data.Category;
    //skill level reference object
    instance.SkillLevel = data.SkillLevel;
}